cmd/kubelet/app/server.go

```diff
@@ -462,10 +717,10 @@ func authPathClientConfig(s *options.KubeletServer, useDefaults bool) (*restclie
 	return &authConfig, nil
 }
 
-func kubeconfigClientConfig(s *options.KubeletServer) (*restclient.Config, error) {
+func kubeconfigClientConfig(kubeConfig string, apiServerList []string) (*restclient.Config, error) {
```

Comment:
I wouldn't change this to use it for bootstrap loading, for a couple reasons:

1. the bootstrap kubeconfig should be self-contained. we don't want to stitch in --api-servers info (we're actually trying to move that direction for --kubeconfig as well, see https://github.com/kubernetes/kubernetes/pull/30798)
2. `NewNonInteractiveDeferredLoadingClientConfig` apparently automatically uses service account credentials if it finds them on disk, and KUBERNETES_HOST envvars as the server

this was a minimal "kubeconfig -> *restclient.Config" loader I came up with:
```

func loadRESTClientConfig(kubeconfig string) (*restclient.Config, error) {
	// Load structured kubeconfig data from the given path
	loader := &clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfig}
	loadedConfig, err := loader.Load()
	if err != nil {
		return nil, err
	}
	// Flatten the loaded data to a particular restclient.Config based on the current context
	return clientcmd.NewNonInteractiveClientConfig(
		*loadedConfig,
		loadedConfig.CurrentContext,
		&clientcmd.ConfigOverrides{},
		loader,
	).ClientConfig()
}
```

I would expect to modify `kubeconfigClientConfig` to look like this eventually, but for now, let's keep it separate and use it for bootstrap purposes